modules/programs/boxxy.nix

```diff
@@ -0,0 +1,110 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.programs.boxxy;
+
+  configPath = "${config.xdg.configHome}/boxxy/boxxy.yaml";
+  settingsFormat = pkgs.formats.yaml { };
+
+  boxxyRulesOpts = types.submodule {
+    freeformType = settingsFormat.type;
+
+    options = {
+      name = mkOption {
+        type = types.str;
+        description = ''
+          Unique identifier of the boxxy rule. This can be any single-line string.
+        '';
+      };
+
+      target = mkOption {
+        type = types.str;
+        default = "";
+        defaultText = "~/.ssh";
+        description = ''
+          What directory/file to redirect.
+        '';
+      };
+
+      rewrite = mkOption {
+        type = types.str;
+        default = "";
+        defaultText = "$XDG_CONFIG_HOME/ssh";
```

Comment:
Did you mean to use `example` here? Also, does variable expansion actually work?

Reply:
Yes, that's the same case as above. I kinda mixed up example and defaultText there.

Reply:
Are you sure? If I'm reading the right part of the source code (https://github.com/queer/boxxy/blob/dd40d5e967c01f4b639a41a1e12c2089b3ab56d5/src/enclosure/fs.rs#L92) it only does tilde expansion, not variable expansion.

Reply:
Oh, apologies. I thought it launches a shell process for each. I should've checked the code as well to be sure.